### PR TITLE
Remove Postgres flag from all packages except Central

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Removes the postgres flag from migrator, tests, roxctl and local

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### CODE REVIEW

**Review by c-du — APPROVED** [review]

**Review by rhybrillou — APPROVED** [review]

File: migrator/clone/clone_test.go [hunk 1]
```diff
@@ -48,30 +48,26 @@ func setVersion(t *testing.T, ver *versionPair) {
 }
 
 func TestCloneMigration(t *testing.T) {
```
Comment (dashrews78) [inline migrator/clone/clone_test.go:50]: With the flag being permanent `TestCloneMigration` and  `TestCloneMigrationRocksToPostgres` are effectively duplicates.  This test and `mock_central.go` need a fair amount of cleanup in this regard.  But all that should be done in a separate PR

File: migrator/main.go [hunk 2]
```diff
@@ -152,29 +148,27 @@ func upgrade(conf *config.Config, dbClone string, processBoth bool) error {
...
 	var gormDB *gorm.DB
 	var pgPool postgres.DB
-	if env.PostgresDatastoreEnabled.BooleanSetting() {
-		pgPool, gormDB, err = postgreshelper.Load(dbClone)
-		if err != nil {
-			return errors.Wrap(err, "failed to connect to postgres DB")
-		}
-		// Close when needed
-		defer postgreshelper.Close()
+	pgPool, gormDB, err = postgreshelper.Load(dbClone)
```
Comment (dashrews78) [inline migrator/main.go:151]: line 128 `if processBoth {`

File: migrator/main.go [hunk 3]
```diff
@@ -152,29 +148,27 @@ func upgrade(conf *config.Config, dbClone string, processBoth bool) error {
...
+	// If Postgres has no version and we have no bolt then we have no populated databases at all and thus don't
+	// need to migrate
+	if ver.SeqNum == 0 && ver.MainVersion == "0" && (!processBoth || boltDB == nil) {
+		log.WriteToStderr("Fresh install of the database. There is no data to migrate...")
+		pkgSchema.ApplyAllSchemas(context.Background(), gormDB)
+		return nil
 	}
+	log.WriteToStderrf("version for %q is %v", dbClone, ver)
 
 	if boltDB == nil && !env.PostgresDatastoreEnabled.BooleanSetting() {
```
Comment (dashrews78) [inline migrator/main.go:173]: I think this one can just go away as it checks to see if we have a legacy DB and in Rocks mode.  Since we can't be in Rocks mode any more we don't need it.

**Comment by dashrews78** [comment]
I took a pass at `clone_test.go` and `mock_central.go` with #6248.  We could either cherry pick that into this one or process the PRs separately.  Either way I would recommend reverting the `clone_test` you have as it has a merge conflict with an other change anyway.
